### PR TITLE
fix(class_hash): add `compiler_version` property to Program

### PR DESCRIPTION
**Pull request by kkovaacs**
This is a new property that was added in Starknet 0.10.0 and contains
the version of the compiler that has been used to compile the program.

We add it as an optional field so that old class definitions can still
be parsed.

### CODE REVIEW

File: crates/pathfinder/src/state/sync/l2.rs [hunk 1]
```diff
@@ -544,6 +544,7 @@ mod tests {
         const DEF0: &str = r#"{
             "abi": [],
             "program": {
+                compiler_version": "0.10.0",
```
Comment (Mirko-von-Leipzig) [inline crates/pathfinder/src/state/sync/l2.rs:547]: I guess this was technically not required? Don't mind it being here, just checking my understanding is correct.

**Review by Mirko-von-Leipzig — APPROVED** [review]
LGTM. Maybe just pending confirmation from SW on the weird [python comments](https://github.com/starkware-libs/cairo-lang/blob/v0.10.0-pre/src/starkware/starknet/core/os/class_hash.py#L105-L108).

Although we can always fixup in a new PR.

**Review by CHr15F0x — APPROVED** [review]

File: crates/pathfinder/src/state/sync/l2.rs (hunk 1 above)
Comment (kkovaacs) [reply crates/pathfinder/src/state/sync/l2.rs:547]: No, you're right, this is technically not required. Shall I remove it?

Comment (Mirko-von-Leipzig) [reply crates/pathfinder/src/state/sync/l2.rs:547]: Maybe, just to make it less confusing when `git blame` in the future.

Comment (kkovaacs) [reply crates/pathfinder/src/state/sync/l2.rs]: Removed. Was syntactically incorrect anyway ;)

**Review by koivunej — APPROVED** [review]
Python comment is interesting but ... I assume this now works?

**Pull request merged**